internal/cmd/server/common.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package server
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+func processBackendToken(
+	ctx context.Context,
+	backendToken string, backendTokenFile string,
+	logger *slog.Logger) (string, error) {
+
+	// Check that the backend token and token file haven't been simultaneously provided:
+	if backendToken != "" && backendTokenFile != "" {
+		logger.ErrorContext(
+			ctx,
+			"Backend token and token file have both been provided, but they are incompatible",
+			slog.Any(
+				"flags",
+				[]string{
+					backendTokenFlagName,
+					backendTokenFileFlagName,
+				},
+			),
+			slog.String("!token", backendToken),
+			slog.String("token_file", backendTokenFile),
+		)
+		return "", errors.New("backendToken and backendTokenFile both provided")
```

Comment:
If we are going to report issues related to flags returning errors then I think they should be more explicit about what caused the error, more similar to what we are writing to the log. For example:

```go
fmt.Errorf(
        "backend token flag '%s' and token file flag '%s' have both been provided, "+
                "but they are incompatible",
        backendTokenFlagName, backendTokenFileFlagName,
)
```

Reply:
Done